cloud-controller-manager/do/cloud.go

```diff
@@ -201,6 +217,28 @@ func (c *cloud) serveDebug(stop <-chan struct{}) {
 	}
 }
 
+func (c *cloud) serveMetrics() {
+	http.Handle("/metrics", promhttp.Handler())
+
+	// register metrics
+	prometheus.MustRegister(apiRequestCounter)
+	prometheus.MustRegister(getRequestCounter)
+	prometheus.MustRegister(createRequestCounter)
+	prometheus.MustRegister(updateRequestCounter)
+	prometheus.MustRegister(successfulRequestCounter)
+	prometheus.MustRegister(failedRequestCounter)
+	prometheus.MustRegister(runLoopsStarted)
+	prometheus.MustRegister(numberOfReconciles)
+	prometheus.MustRegister(createRequestDuration)
+	prometheus.MustRegister(getRequestDuration)
+	prometheus.MustRegister(updateRequestDuration)
+	prometheus.MustRegister(eventHandlerDuration)
+
+	if err := http.ListenAndServe(c.metrics.host, nil); err != http.ErrServerClosed {
```

Comment:
I suppose this will fail if `c.metrics.host` is empty, which AFAICS would be legitimate in case metrics are disabled. We should gate the entire metrics setup in this method on the host variable being non-empty.